module/zfs/vdev_queue.c

```diff
@@ -349,14 +375,34 @@ vdev_queue_class_max_active(spa_t *spa, zio_priority_t p)
 	case ZIO_PRIORITY_ASYNC_WRITE:
 		return (vdev_queue_max_async_writes(spa));
 	case ZIO_PRIORITY_SCRUB:
+		if (vq->vq_ia_active > 0) {
+			return (MIN(vq->vq_nia_credit,
+			    zfs_vdev_scrub_min_active));
+		} else if (vq->vq_nia_credit < zfs_vdev_nia_delay)
+			return (zfs_vdev_scrub_min_active);
```

Comment:
I'm a little confused about what vq_nia_credit is, conceptually.  Is it 2 different ideas depending on whether `vq_ia_active == 0`?

Reply:
Yes.  When vq_ia_active == 0, it counts non-interactive requests since last completed interactive one to decide when to allow max_active (zfs_vdev_nia_delay).  When vq_ia_active > 0, it counts sent non-interactive requests since last completed interactive.